autonick.lua

```diff
@@ -0,0 +1,56 @@
+-- automatically nickname dwarves
+--[====[
+
+autonick
+========
+Automatically nickname dwarves according to ``dfhack-config/autonick.txt``.
+
+One nickname per line.
+Empty lines, lines beginning with ``#`` and repeat entries are discarded.
+
+Dwarves with manually set nicknames are ignored.
+
+If there are fewer available nicknames than dwarves, the remaining
+dwarves will go un-nicknamed.
+
+You may wish to use this script with the "repeat" command, e.g:
+``repeat -name autonick -time 3 -timeUnits months -command [ autonick ]``
+
+]====]
+
+names = {};
+
+path = dfhack.getDFPath () .. "/dfhack-config/autonick.txt";
+
+-- grab list, put in array
+for line in io.lines(path) do
+    if (line:trim() ~= "") or (line:sub(1, 1) == "#") then
+        table.insert(names, line);
+    end;
+end;
+
+--check current nicknames
+for _,unit in ipairs(df.global.world.units.active) do
+    if dfhack.units.isCitizen(unit) and
+    unit.name.nickname ~= "" then
+        for i,name in ipairs(names) do
+            if name == unit.name.nickname then
+                --remove current nickname from array
+                table.remove(names, i);
```

Comment:
Instead of looping through to remove taken nicknames, how about filtering them out when we build the names list?

You could add this before the io.lines loop above (sorry for the formatting.. I'm writing this on my phone):

local taken = {}
for _,unit in ipairs(df.global.world.units.active) do
    if dfhack.units.isCitizen(unit) and
    unit.name.nickname ~= "" then taken[unit.name.nickname] = true end
end

And then when building names, check if taken[name] is truthy

Reply:
will do this next.